OneCommunityGlobal/HighestGoodNetworkApp
# Shiwani add permission for edit project category or status

# Description
Make permission management fix for edit project category or status.

## Related PRS (if any):
This frontend PR is related to [#650](https://github.com/OneCommunityGlobal/HGNRest/pull/650) backend PR.

## Main changes explained:
- Added anEditCategoryAndStatus to Project.jsx which will check if 'editProject' permission is present in frontend permissions. If it is true then category and status field will be editable.
- Added canEditCategoryAndStatus as a condition in return to make it editable.


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner user
4. go to dashboard→ Other links -> Permission Management -> Manage user permissions -> type name of any user -> Add 'Edit project Category or Status' -> click submit
5. If the user for whom the permission is being set does not have projects tab under Other links then add 'See Project Management Tab (Add team members, upload new WBSs only)' permission as well for that user
6. Log in as the user for whom the permission was added
7. go to dashboard -> Other links -> Projects -> check if category and status is editable.
8. Check if name is not editable, there are no 403 errors in browser console when trying to edit category or status.

## Screenshots or videos of changes:
Before:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/72ce48ed-e3c4-4366-a34d-64fc0d85059b



After:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/65c5e668-b1d9-46e7-be2a-b5dcc1fa4ff9

## Note
- Owner, Administrator, Manager have permission to edit name, category, status as there are preset permissions for these roles that allow them to be edited by default.
- Check for roles that do not have category or status edit permission like volunteer, assistant manager, test or any other role.
